src/ck-perf/trace-common.C

```diff
@@ -261,14 +261,16 @@ extern const char* const CmiCommitID;
 /** Write out the common parts of the .sts file. */
 void traceWriteSTS(FILE *stsfp,int nUserEvents) {
   fprintf(stsfp, "MACHINE \"%s\"\n",CMK_MACHINE_NAME);
-#if CMK_SMP_TRACE_COMMTHREAD
+#if CMK_SMP_TRACE_COMMTHREAD && !CMK_SMP_NO_COMMTHD
```

Comment:
Not to keep causing churn because of this one line, but after thinking more I think this would be more pedantically correct.
```suggestion
#if CMK_SMP_TRACE_COMMTHREAD && CMK_SMP && !CMK_SMP_NO_COMMTHD
```

Reply:
Looks good to me, but what case is this meant to capture that the other defines would not?

Reply:
Yes, exactly that.

Reply:
Gotcha, makes sense.